server/jetstream_api.go

```diff
@@ -3323,12 +3348,14 @@ func (s *Server) jsConsumerInfoRequest(sub *subscription, c *client, subject, re
 		if !acc.JetStreamIsConsumerLeader(streamName, consumerName) {
 			if js.isLeaderless() {
 				resp.Error = jsClusterNotAvailErr
-				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+				// Delaying an error response gives the leader a chance to respond before us
+				s.sendDelayedAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp), ca.Group)
 			} else if rg := ca.Group; rg != nil && rg.node != nil && rg.isMember(cc.meta.ID()) {
 				// Check here if we are a member and this is just a new consumer that does not have a leader yet.
 				if rg.node.GroupLeader() == _EMPTY_ && !rg.node.HadPreviousLeader() {
 					resp.Error = jsNoConsumerErr
-					s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+					strMsg := string(msg)
```

Comment:
Missed one here and you are calling sendAPIErrResponse, not the delayed one..

Reply:
fixed